test/checkpoint/CMakeLists.txt

```diff
@@ -33,10 +33,15 @@ create_test_executable(test_checkpoint
         checkpointer.c
         workers.c
         test_checkpoint.c
-    # smoke.sh expects the output binary name is 't'.
-    EXECUTABLE_NAME "t"
-    ADDITIONAL_FILES ${CMAKE_CURRENT_SOURCE_DIR}/smoke.sh
 )
 
-add_test(NAME test_checkpoint COMMAND ${CMAKE_CURRENT_BINARY_DIR}/smoke.sh)
-set_tests_properties(test_checkpoint PROPERTIES LABELS "check")
+define_test_variants(test_checkpoint
+    VARIANTS
+        "test_checkpoint_6_row;-T 6 -t r"
+        "test_checkpoint_6_row_named;\"-c TeSt\" -T 6 -t r"
```

Comment:
What's with the escaped quotes around `-c TeSt`?

Reply:
Thanks for bringing this up. I initially did it to get around a problem where `-c` and `TeSt` where interpreted as separate arguments, described by CMake here: https://cmake.org/cmake/help/latest/command/separate_arguments.html
> A Windows command-line is parsed using the same syntax the runtime library uses to construct argv at startup. It separates arguments by whitespace that is not double-quoted.
I've since refactored the helper a bit since then and testing now it seems to no longer be an issue that occurs. I've removed the escaped quotes.